### PR TITLE
Benchmarks: Model benchmarks - Add log for each step

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
add log every steps for model benchmarks.

**Major Revision**
- add log_every_steps option to all benchmarks
- print step time to stdout every n steps

### CODE REVIEW

File: superbench/benchmarks/model_benchmarks/model_base.py [hunk 1]
```diff
@@ -131,6 +131,14 @@ def add_parser_arguments(self):
             help='Enable option to use full float32 precision.',
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/model_base.py:131]: Do we need to add in document?

File: superbench/benchmarks/model_benchmarks/pytorch_bert.py [hunk 2]
```diff
@@ -148,6 +148,9 @@ def _train_step(self, precision):
                 self._optimizer.step()
                 end = self._timer()
                 curr_step += 1
+                if self._args.log_every_steps:
+                    if curr_step % self._args.log_every_steps == 0:
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/pytorch_bert.py:152]:
```
if self._args.log_every_steps && curr_step % self._args.log_every_steps == 0:
```

File: superbench/benchmarks/model_benchmarks/pytorch_bert.py [hunk 3]
```diff
@@ -176,6 +179,11 @@ def _inference_step(self, precision):
                     self._model(sample)
                     end = self._timer()
                     curr_step += 1
+                    if self._args.log_every_steps:
+                        if curr_step % self._args.log_every_steps == 0:
+                            print(
+                                f'{self._name} - {precision.value}: step {curr_step}, step time {(end - start) * 1000}'
+                            )
```
Comment (guoshzhao) [inline superbench/benchmarks/model_benchmarks/pytorch_bert.py:186]: Could we put these code into function, and call function directily?

File: superbench/benchmarks/model_benchmarks/pytorch_bert.py (hunk 2 above)
Comment (yukirora) [reply superbench/benchmarks/model_benchmarks/pytorch_bert.py]: done

File: superbench/benchmarks/model_benchmarks/pytorch_bert.py (hunk 3 above)
Comment (yukirora) [reply superbench/benchmarks/model_benchmarks/pytorch_bert.py]: sure, updated

**Review by guoshzhao — APPROVED** [review]

File: superbench/benchmarks/model_benchmarks/model_base.py [hunk 4]
```diff
@@ -435,3 +443,15 @@ def print_env_info(self):
...
+        """Log step time into stdout regularly.
+
+        Args:
+            curr_step (int): the index of the current step
+            precision (Precision): precision of model and input data, such as float32, float16.
+            start (float): the start timestamp of the current step
+            end (float): the end timestamp of the current step
+        """
+        if self._args.log_every_steps and curr_step % self._args.log_every_steps == 0:
+            print(f'{self._name} - {precision.value}: step {curr_step}, step time {(end - start) * 1000}')
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/model_base.py:457]: Shall we use unified log method? I search the code base, there is no `print` in the main codebase

File: superbench/benchmarks/model_benchmarks/model_base.py [hunk 5]
```diff
@@ -131,6 +131,14 @@ def add_parser_arguments(self):
             help='Enable option to use full float32 precision.',
         )
 
+        self._parser.add_argument(
+            '--log_every_steps',
```
Comment (abuccts) [inline superbench/benchmarks/model_benchmarks/model_base.py:135]: log_n_steps?

File: superbench/benchmarks/model_benchmarks/pytorch_bert.py [hunk 6]
```diff
@@ -176,6 +177,7 @@ def _inference_step(self, precision):
                     self._model(sample)
                     end = self._timer()
                     curr_step += 1
+                    self._log_step_time(curr_step, precision, start, end)
                     if curr_step > self._args.num_warmup:
                         # Save the step time of every training/inference step, unit is millisecond.
                         duration.append((end - start) * 1000)
```
Comment (abuccts) [inline superbench/benchmarks/model_benchmarks/pytorch_bert.py:183]: maybe log the average of last n steps in duration?

**Pull request closed without merging**